File: test-web-search.py
import openai
import json
import os

client = openai.OpenAI(
    api_key=os.environ.get("OPENAI_API_KEY")
)

print("Testing OpenAI web search with gpt-5-mini...")
print("=" * 60)

try:
    response = client.responses.create(
        model="gpt-5-mini",
        input=[
            {
                "role": "developer",
                "content": [{"type": "input_text", "text": "You are a market research analyst. Search the web for information and return a brief JSON summary."}],
            },
            {
                "role": "user",
                "content": [{"type": "input_text", "text": "Search the web for information about the chiral molecules analytical tools market size and key players. Return a brief JSON with summary, competitors array, and marketSize."}],
            },
        ],
        tools=[
            {
                "type": "web_search_preview",
                "search_context_size": "high",
                "user_location": {
                    "type": "approximate",
                    "country": "FI",
                },
            }
        ],
        text={"format": {"type": "text"}},
        store=False,
    )

    print("SUCCESS! Response received.")
    print(f"Output type: {type(response)}")
    
    # Extract text
    if hasattr(response, 'output') and response.output:
        for item in response.output:
            if hasattr(item, 'type') and item.type == 'message' and hasattr(item, 'content'):
                for c in item.content:
                    if hasattr(c, 'type') and c.type == 'output_text':
                        print(f"\nResponse text (first 500 chars):\n{c.text[:500]}")
    
    if hasattr(response, 'output_text') and response.output_text:
        print(f"\noutput_text (first 500 chars):\n{response.output_text[:500]}")

except Exception as e:
    print(f"ERROR: {type(e).__name__}: {e}")
    import traceback
    traceback.print_exc()

File: test-web-search.mjs
// Test web search with OpenAI - matching the working market-analysis pattern
import OpenAI from 'openai';
import { readFileSync } from 'fs';

// Parse .env.local manually
const envContent = readFileSync('.env.local', 'utf-8');
for (const line of envContent.split('\n')) {
    const [key, ...vals] = line.split('=');
    if (key && vals.length) process.env[key.trim()] = vals.join('=').trim();
}

async function test() {
    console.log('API Key:', process.env.OPENAI_API_KEY?.slice(0, 20) + '...');
    console.log('Model:', process.env.OPENAI_MODEL || 'gpt-5.4');
    console.log('');

    const openai = new OpenAI({ apiKey: process.env.OPENAI_API_KEY });

    console.log('Test 1: Simple web search (matching market-analysis pattern)...');
    try {
        const response = await openai.responses.create({
            model: process.env.OPENAI_MODEL || 'gpt-5.4',
            input: [
                {
                    role: 'developer',
                    content: [{ type: 'input_text', text: 'You are a market research analyst. Be brief.' }],
                },
                {
                    role: 'user',
                    content: [{ type: 'input_text', text: 'Search the web for the current market size of chiral chromatography. Reply in 2 sentences.' }],
                },
            ],
            reasoning: {
                effort: 'medium',
                summary: 'auto',
            },
            text: { format: { type: 'text' } },
            tools: [
                {
                    type: 'web_search_preview',
                    search_context_size: 'medium',
                    user_location: { type: 'approximate', country: 'FI' },
                },
            ],
            store: false,
        });

        console.log('SUCCESS!');
        if (response.output_text) {
            console.log('Response:', response.output_text.slice(0, 300));
        }
    } catch (e) {
        console.log('FAILED:', e.constructor.name, e.message);
        if (e.cause) console.log('Cause:', e.cause.message || e.cause);
    }
}

test();
